src/cmd/link/internal/ld/pcln.go

```diff
@@ -249,6 +249,16 @@ func (state *pclntab) generatePCHeader(ctxt *Link) {
 		if off != size {
 			panic(fmt.Sprintf("pcHeader size: %d != %d", off, size))
 		}
+
+		if val, ok := os.LookupEnv("GARBLE_LINKER_MAGIC"); ok {
```

Comment:
how about:

```
magicStr := os.Getenv("GARBLE_LINKER_MAGIC")
if magicStr == "" {
    panic(we expect this to always be set for garble builds)
}
rest of your code...
```

this way we also ensure we fail if, for some reason, the magic value isn't set or is empty

Reply:
Makes sense, this patch was designed to support enabling/disabling magic value randomization, and now it is no longer needed